src/components/ToastNotifications.tsx

```diff
@@ -0,0 +1,193 @@
+      setFilters({});
+    }
+  };
+
+  const handleDismissGroups = () => {
+    if (hasFilters) {
+      onDismissByFilter(notifications, filters);
+      setFilters({});
+      return;
+    }
+
+    onDismissAll();
+  };
+
+  const getSeverityFilters = () => {
+    const filterButtons = severityOrder.map((severity) => {
+      if (groupedCount[severity]) {
+        return (
+          <button
+            aria-label={`Filter ${severity} notifications`}
+            aria-pressed={!!filters[severity]}
+            key={severity}
+            className={classnames(
+              "u-no-margin u-no-padding u-no-border filter-button",
+              { "filter-button-active": filters[severity] },
+            )}
+            onClick={() => handleFilterSelect(severity)}
+            disabled={!expanded}
+          >
+            <Icon name={iconLookup[severity]} />
+            <span>{groupedCount[severity]}</span>
+          </button>
+        );
+      }
+    });
+
+    return <div className="filters">{filterButtons}</div>;
+  };
+
+  const getDismissText = () => {
+    if (hasFilters) {
+      const dismissText = (
+        <div className="filters">
+          <span className="filter-text">Dismiss</span>
+          {severityOrder.map((severity) => {
+            if (groupedCount[severity] && filters[severity]) {
+              return (
+                <span key={`dismiss-text-${severity}`} className="filter-text">
+                  <span>{groupedCount[severity]}</span>
+                  <Icon name={iconLookup[severity]} />
+                </span>
+              );
+            }
+          })}
+        </div>
+      );
+      return dismissText;
+    }
+
+    return <span>Dismiss all {notifications.length}</span>;
+  };
+
+  // Only filter input data if there are filters present
+  const filteredNotifications = hasFilters
+    ? notifications.filter((notification) => filters[notification.type])
+    : notifications;
+
+  const lastNotificationIndex = filteredNotifications.length - 1;
+  const latestNotification = filteredNotifications[lastNotificationIndex];
+  // Don't assign alert role for notifications when expanded since we don't want
+  // screen readers to announce every existing notification
+  const notificationEls = expanded ? (
+    filteredNotifications.map((_, index, array) => {
+      const lastNotificationIndex = array.length - 1;
+      // This will map notifications in reverse order
+      const notification = array[lastNotificationIndex - index];
+      return (
+        <Notification
+          key={`notification-${index}`}
+          title={notification.title ?? DefaultTitles[notification.type]}
+          actions={notification.actions}
+          severity={notification.type}
+          onDismiss={() => onDismiss(notification)}
+          className={`u-no-margin--bottom toast-notification ${index === lastNotificationIndex ? "toast-notification--no-border-bottom" : ""}`}
+          timestamp={notification.timestamp}
```

Comment:
We probably should set `titleElement` to a non heading one like `div`, because the default of `h5` will not comply to the document heading structure.

Reply:
yeah in this case I think `div` is appropriate since toast notifications exists outside of the app root.